spec/helper.ts

```diff
@@ -71,29 +72,45 @@ export function runHandler(
       }
 
       public send(sendBody: any) {
-        const toSend = typeof sendBody === "object" ? JSON.stringify(sendBody) : sendBody;
-        const body = this.sentBody ? this.sentBody + ((toSend as string) || "") : toSend;
-
-        resolve({
-          status: this.statusCode,
-          headers: this.headers,
-          body,
-        });
-        if (this.callback) {
-          this.callback();
+        if (this.writeCalled) {
+          throw Error("Cannot set headers after they are sent to the client");
         }
+
+        const toSend = typeof sendBody === "object" ? JSON.stringify(sendBody) : sendBody;
+        const body =
+          typeof this.sentBody === "undefined"
+            ? toSend
+            : this.sentBody + ((toSend as string) || "");
```

Comment:
nit: consider String(toSend || "") if you want to both silence errors _and_ have defined behavior at runtime